udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -31,23 +30,55 @@
   public static final int OVERLAPS_START = 4;
   public static final int OVERLAP_CONTAINS = 5;
 
+  /**
+   * Returns a collection of TimeRange objects to indicate available meeting times
+   * for indicated people and duration given through a request object.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    /* Create an output collection with a full day to cut(shorten or split)
+
+    // Perform queries with and without optional attendees.
+    Collection<TimeRange> mandatoryCheck = coreQuery(events, request, false);
+    Collection<TimeRange> optionalCheck = coreQuery(events, request, true);
+
+    // Return optional attendee checks if valid.
+    if ((optionalCheck.size() != 0) || (request.getAttendees().size() == 0)) {
```

Comment:
Do we need to check if getOptionalAttendees().size > 0? In other words, does it make sense to return the optionalCheck range if there were no attendees marked as optional?

Reply:
If there are no optional attendees then returning the optionalCheck is still the valid return as the algorithm of the optional check is the same as mandatoryCheck but including optional attendees. If there are none, optionalCheck is equivalent to mandatoryCheck.